parser.y

```diff
@@ -3123,6 +3123,13 @@ DatabaseOption:
 	}
 |	DefaultKwdOpt "ENCRYPTION" EqOpt stringLit
 	{
+		if "Y" == strings.ToUpper($4) {
+			yylex.AppendError(yylex.Errorf("The ENCRYPTION clause is parsed but ignored by all storage engines."))
+			parser.lastErrorAsWarn()
+		} else if "N" != strings.ToUpper($4) {
+			yylex.AppendError(ErrWrongValue.GenWithStackByArgs("argument (should be Y or N)", $4))
+			return 1
+		}
```

Comment:
Better use a `switch`/`case` here (maybe get rid of `strings.ToUpper` too), or create a dedicated rule for `'Y'`/`'y'`/`'N'`/`'n'` in place of `stringLit` (since the same also applies to TableOption).

```go
switch $4 {
case "Y", "y":
    yylex.AppendError(...)
    parser.lastErrorAsWarn()
case "N", "n":
    break
default:
    yylex.AppendError(...)
    return 1
}
```

Reply:
I have modified the implementation of the code.